swingset-demo/src/main/java/com/nqadmin/swingset/demo/DemoUtil.java

```diff
@@ -22,14 +23,236 @@
+ * Two of the ways allow the assignment of a connection to a JdbcRowSet,
+ * several JdbcRowSet can share a single connection;
+ * any threading and/or concurancy issues are handled by the connection.
+ * The third way hooks into an actual connection pool, which is provided
+ * by the database, and uses a {@link javax.sql.rowset.CachedRowSet}; in the case of the demo
+ * the pool is never bigger than one.
+ * <p>
+ * One way uses {@link com.nqadmin.rowset.JdbcRowSetImpl}, which is a custom
+ * JdbcRowSet implementation, which allows the direct assignment of a connection
+ * to the RowSet for its use. This is useful in environments where a
+ * {@link RowSetProvider} can not be used. For example
+ * <pre>
+ * {@code 
+ *    rowSet = new JdbcRowSetImpl(connection);
+ * }
+ * </pre>
+ * <p>
+ * The demo contains two different DataSource implementations. One is trivial
+ * and allows a given connection to be associated with a factory provided
+ * RowSet. This scheme is semantically equivelent to the previously mentioned
+ * mechanism, but without requiring a custom RowSet implementation.
+ * It is used like
+ * <pre>
+ * {@code 
+ *    rs = RowSetProvider.newFactory().createJdbcRowSet();
+ *    rs.setDataSourceName(getDsName(connection));
+ * }
+ * </pre>
+ * <p>
+ * Here is an example for the third way using connection pool,
+ * based on a connection pool provided by the database.
+ * <pre>
+ * {@code 
+ *    rs = RowSetProvider.newFactory().createCachedRowSet();
+ *    rs.setDataSourceName(DataSourcePool.DATA_SOURCE_NAME);
+ * }
+ * </pre>
  */
-public class DemoUtil
-{ 
-    private static final Logger logger = LogManager.getLogger(MainClass.class);
+public class DemoUtil { 
+	private DemoUtil() { }
+	private static final Logger logger = LogManager.getLogger(MainClass.class);
+
+	/** Invoke this method to output collected statistics to log. */
+	public static void logConnectionUsage() {
+		logger.info(() -> "RowSetSourceDefault: " + whichRowSetDefault);
+		logger.info(() -> String.format(
+				"Connection pool: max active: %d, nOpen %d, nClose %d",
+				DataSourcePool.cMax(), DataSourcePool.nOpen(), DataSourcePool.nClose()));
+		for (String dsName : connMap.values()) {
+			logger.info(() -> "Connection dsName: " + dsName);
+		}
+	}
+
+	/**
+	 * Used to specify how to create a new {@linkplain RowSet}.
+	 */
+	public enum RowSetSource {
+		/** Use specified connection and JdbcRowSet. */
+		SHARE_JDBC,
+		/** Use normal connection pool and CachedRowSet. */
+		POOL_CACHED,
+		/** Use custom JdbcRowSet. */
+		NQADMIN,
+	}
+	/** Track every connection that's seen. If a connection does not have an
+	 * associated DataSourceName then the value is null.
+	 */
+	private static final Map<Connection, String> connMap = new IdentityHashMap<>();
+	private static InitialContext initialContext = null;
+	private static RowSetFactory rsFactory = null;
+
+	private static RowSetSource whichRowSetDefault = RowSetSource.SHARE_JDBC;
```

Comment:
Did we decide in a comment or email that we want to use NQADMIN as the default?

Reply:
Not that I recall. This default is a JdbcRowSet (JdbcRowSetImpl is also a JdbcRowSet)